src/view/widgets/multi_select_widget.py

```diff
@@ -304,24 +374,67 @@ def load_options_after_filter_got_set(
         self, target_widget_name, category, filter_widget, event
     ):
         """
-        Loads options for item filter after upper item filter was triggered. First select the referenced item filter by name
-        and then get and insert new filter options for the target item filter based on the selected upper item filter.
-
-        :param target_widget_name: target item filter widgets name.
-        :param category: category to set how to get new option values for target item filter.
-        :param filter_widget: upper item filter widget.
-        :param event: event type triggered by upper item filter.
+        Updates options for the target item filter when the upper item filter is triggered.
         """
-        if (event.type != "changed") or (category != "genres"):
-            return
+        # upper_category = equsls to the widget that is named filter_widget
+        # widget = equsls to the widget that has the named category
 
-        target_widget = find_widget_by_name_recursive(
+        # Find the target widget (item filter widget) by name
+        target_widget = self.find_widget_by_name_recursive(
             self.reco_explorer_app_instance.config_based_nav_controls,
             target_widget_name,
         )
 
+        print("the widget is ")
+        print(target_widget)
+
         target_widget.value = (
             self.controller_instance.get_genres_and_subgenres_from_upper_category(
                 filter_widget.value, category
             )
         )
+
+        print("the widget is ")
```

Comment:
logging would be more appropriate if this is relevant information, if these are for debugging purposes, please remove them

Reply:
Done.